### ISSUE TITLE
Upgrade to openshift 1.4.0+ breaks downloading caching 'oc' binary

### ISSUE THREAD

**Issue by budhrg** [minishift/minishift#331]
It looks like regression of upgrading to `1.4.0`, `minishift start` fails on caching `oc` binary.

```
minishift start --vm-driver virtualbox
Starting local OpenShift cluster using 'virtualbox' hypervisor...
Downloading ISO 'https://github.com/minishift/minishift-b2d-iso/releases/download/v1.0.0/minishift-b2d.iso'
 40.00 MB / 40.00 MB [============================================================================================================================] 100.00% 0s
 22.37 MB / 22.37 MB [============================================================================================================================] 100.00% 0s
E0125 10:46:57.786410   26875 start.go:219] Error starting the cluster:  Error attempting to download and cache oc: Cannot untar /tmp/minishift-asset-download-280664917/openshift-origin-client-tools-v1.4.0-208f053-linux-64bit.tar: open /tmp/minishift-asset-download-280664917/openshift-origin-client-tools-v1.4.0+208f053-linux-64bit/oc: no such file or directory
```

Found that now from 1.4.1 onwards, tar directory `openshift-origin-client-tools-v1.4.0+208f053-linux-64bit` includes `+` instead of `-` which is not the case in windows or mac.

Might be regression from origin itself and filing issue there.

**Comment by hferentschik** [comment]
See also https://github.com/openshift/origin/issues/12663

**Comment by gbraad** [comment]
I was wondering if the `untar` library would support disregarding the folder structure. With GNU `tar` you can use `--strip=1` that would achieve such a thing.

**Comment by hferentschik** [comment]
The library in fact only iterates files. It is up to you to create the directory structure if you want. one could flatten as well. The key is to just not make any assumptions on the directory structure as we did before.

**Comment by hferentschik** [comment]
Merged via pull request #333
